clock/timer.go

```diff
@@ -56,13 +53,20 @@ func (t *Timer) Reset(d time.Duration, tags ...string) bool {
 	t.mock.matchCallLocked(c)
 	defer close(c.complete)
 	result := !t.stopped
-	t.mock.removeTimerLocked(t)
-	t.stopped = false
-	t.nxt = t.mock.cur.Add(d)
 	select {
 	case <-t.c:
 	default:
 	}
+	if d == 0 {
```

Comment:
before, when we had `Advance()` handle multiple sets of timers/ticks events, we just handled this case of a zero-duration timer reset in the advance loop.  But now, since `Advance()` can only trigger one "flight" of events, we handle the zero-duration case inline.

One slight disadvantage is that there isn't a way to wait for such events to complete.